Update azure documentation (#3611)
open-cluster-management/rhacm-docs: add-ons/submariner/submariner_prepare_hosts.adoc

To prepare the clusters on your Microsoft Azure for deploying the Submariner component, complete the following steps:

> . Tag a node as a gateway node.

Reviewer: For all the other cloud providers with ACM, we are recommending a dedicated GW node. Should we do the same even for Azure?

Author: This is a temporary solution for trying in tech preview , isn't creating machine set complicated ?

+
----
kubectl label nodes <worker-node-name> "submariner.io/gateway=true" --overwrite
----
+
. Create a public IP and assign it to the VM of the node that was tagged as gateway node
+
----
az network public-ip create --name <public-ip-name> --resource-group <res-group>

> az network nic ip-config update --name <name>  --nic-name <gw-vm-nic> --resource-group <res-group>  --public-ip-address <public-ip-name>

Reviewer: It would be good to provide some brief description below all the commands to explain about the argument that needs to be supplied as part of the command.

Author: Give a description for arguments that need explanation.

----
+
. Create a network security group for submariner gateway
+
----

> az network nsg create --name <gw-nsg-name> --resource-group <rescource group>

Reviewer: ```suggestion
az network nsg create --name <gw-nsg-name> --resource-group <res-group>
```

Lets try to use the same names for readability.

----
+

> . Create network security groups rules on your Azure to open NAT traversal ports (by default 4500/UDP) Nat discovery port (by default 4490/UDP) and metrics port (by default 8080/TCP) for Submariner. This needs to be created in both inbound and outbound directions for each of the ports.

Reviewer: ```suggestion
. Create network security groups rules on your Azure to open Tunnel port (by default 4500/UDP), NAT discovery port (by default 4490/UDP) and metrics port (by default 8080/TCP and 8081/TCP) for Submariner. This needs to be created in both inbound and outbound directions for each of the ports.
```

+
----
az network nsg rule create --resource-group <res-group> \
--nsg-name <gw-nsg-name> --priority <priority> \
--name <name> --direction Inbound --access Allow \
--protocol <Protocol> --destination-port-ranges <port>

az network nsg rule create --resource-group <res-group> \
--nsg-name <gw-nsg-name> --priority <priority> \
--name <name> --direction Outbound --access Allow \
--protocol <Protocol> --destination-port-ranges <port>
----
+
. Attach the security group to the gateway VM interface
+
----
az network nic update -g <res-group> -n <gw-vm-nic>  --network-security-group <gw-nsg-name>
----
+
. Create network security groups rules on your Azure to open VXLAN port (by default 4800/UDP) on the existing security groups(<resource-group-name>-nsg by default) that is associated with the worker and the master
----
az network nsg rule create --resource-group <res-group> \
--nsg-name <nsg-name> --priority <priority> \
--name <name> --direction Inbound --access Allow \

> --protocol <Protocol> --destination-port-ranges <ipsec-port>

Reviewer: ```suggestion
--protocol udp --destination-port-ranges <vxlan-port>
```

Just like how we are using proper values for direction and some of the other fields, I feel we can use the protocol values which do not change in our examples.

az network nsg rule create --resource-group <res-group> \
--nsg-name <nsg-name> --priority <priority> \
--name <name> --direction Outbound --access Allow \

> --protocol <Protocol> --destination-port-ranges <ipsec-port>

Reviewer: Same as above.
